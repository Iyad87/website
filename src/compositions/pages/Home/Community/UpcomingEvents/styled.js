import styled from 'styled-components';
import { textLight } from 'styles/colours';
import { spacing } from 'styles/spacing';
import breakpoints from 'styles/breakpoints';

export const Events = styled.ul`
  display: flex;
  flex-wrap: wrap;
  list-style-type: none;
  margin: ${spacing(-1.25)};
  margin-bottom: ${spacing(6.25)};
  padding: 0;
`;

export const Event = styled.li`
  display: block;
  padding: ${spacing(1.25)};
  width: 100%;

  ${breakpoints.tablet`
    width: 50%;
  `};

  ${breakpoints.desktop`
    width: ${100 / 3}%;
  `};
`;

export const Title = styled.h3`
  color: ${textLight};
  font-weight: 500;
  font-size: 2.25rem;
  margin-top: 0;
  margin-bottom: ${spacing(5)};
`;
