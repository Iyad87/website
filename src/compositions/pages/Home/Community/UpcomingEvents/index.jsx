import React from 'react';
import Post from 'components/Post';
import { Event, Events, Title } from './styled';

import Button from 'components/Button';
import Link from 'components/Link';

const UpcomingEvents = () => (
  <section>
    <header>
      <Title>Upcoming Events</Title>
    </header>
    <Events>
      <Event>
        <Post
          date="2018-05-18T14:38:23.996+01:00"
          image="https://placekitten.com/128/128"
          sourceLink="/events/digital-product-london"
          sourceText="Digital Product London"
          title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
        >
          I think its one of the simplest yet most powerful extensions inVision
          hasn&apos;t launched.
        </Post>
      </Event>
      <Event>
        <Post
          date="2018-05-18T14:38:23.996+01:00"
          image="https://placekitten.com/128/128"
          sourceLink="/events/digital-product-london"
          sourceText="Digital Product London"
          title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
        >
          I think its one of the simplest yet most powerful extensions inVision
          hasn&apos;t launched.
        </Post>
      </Event>
      <Event>
        <Post
          date="2018-05-18T14:38:23.996+01:00"
          image="https://placekitten.com/128/128"
          sourceLink="/events/digital-product-london"
          sourceText="Digital Product London"
          title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
        >
          I think its one of the simplest yet most powerful extensions inVision
          hasn&apos;t launched.
        </Post>
      </Event>
    </Events>
    <Button component={Link} href="/events">
      View Events
    </Button>
  </section>
);

export default UpcomingEvents;
