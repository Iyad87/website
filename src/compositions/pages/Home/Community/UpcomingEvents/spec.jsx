import React from 'react';
import { shallow } from 'enzyme';
import UpcomingEvents from './index';

test('compositions/Home/Community/UpcomingEvents snapshot', () => {
  const wrapper = shallow(<UpcomingEvents />);

  expect(wrapper).toMatchSnapshot();
});
