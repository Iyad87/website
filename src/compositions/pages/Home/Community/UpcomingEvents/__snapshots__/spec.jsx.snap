// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compositions/Home/Community/RecentBlogPosts snapshot 1`] = `
<React.Fragment>
  <header>
    <styled.h3>
      Recent Blog Posts
    </styled.h3>
  </header>
  <styled.ul>
    <styled.li>
      <Post
        date="2018-05-18T14:38:23.996+01:00"
        image="https://placekitten.com/128/128"
        sourceLink="/people/nuno-job"
        sourceText="Nuno Job"
        title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
      >
        I think its one of the simplest yet most powerful extensions inVision hasn't launched. As animation and transition in inVision evolve it should grow even more powerful
      </Post>
    </styled.li>
    <styled.li>
      <Post
        date="2018-05-18T14:38:23.996+01:00"
        image="https://placekitten.com/128/128"
        sourceLink="/people/nuno-job"
        sourceText="Nuno Job"
        title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
      >
        I think its one of the simplest yet most powerful extensions inVision hasn't launched. As animation and transition in inVision evolve it should grow even more powerful
      </Post>
    </styled.li>
  </styled.ul>
  <Button
    component={[Function]}
    href="/blog"
  >
    View Blog
  </Button>
</React.Fragment>
`;

exports[`compositions/Home/Community/UpcomingEvents snapshot 1`] = `
<section>
  <header>
    <styled.h3>
      Upcoming Events
    </styled.h3>
  </header>
  <styled.ul>
    <styled.li>
      <Post
        date="2018-05-18T14:38:23.996+01:00"
        image="https://placekitten.com/128/128"
        sourceLink="/people/nuno-job"
        sourceText="Digital Product London"
        stretch={true}
        title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
      >
        I think its one of the simplest yet most powerful extensions inVision hasn't launched.
      </Post>
    </styled.li>
    <styled.li>
      <Post
        date="2018-05-18T14:38:23.996+01:00"
        image="https://placekitten.com/128/128"
        sourceLink="/people/nuno-job"
        sourceText="Digital Product London"
        stretch={true}
        title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
      >
        I think its one of the simplest yet most powerful extensions inVision hasn't launched.
      </Post>
    </styled.li>
    <styled.li>
      <Post
        date="2018-05-18T14:38:23.996+01:00"
        image="https://placekitten.com/128/128"
        sourceLink="/people/nuno-job"
        sourceText="Digital Product London"
        stretch={true}
        title="We’re Hiring! YLD on the look out for Front-End Devs and DevOps."
      >
        I think its one of the simplest yet most powerful extensions inVision hasn't launched.
      </Post>
    </styled.li>
  </styled.ul>
  <Button
    component={[Function]}
    href="/blog"
  >
    View Blog
  </Button>
</section>
`;
